tests/jobbergate/test_submit.py

```diff
@@ -415,3 +415,53 @@ def _submit_side_effect(request):
         ).encode("utf-8")
 
         assert update_3_route.call_count == 1  # called to notify the job was rejected
+
+
+class TestGetJobParameters:
+    """
+    Test the ``get_job_parameters()`` function.
+    """
+
+    def test_base_case__fail(self):
+        """
+        Base case should fail, since name is a required field.
+        """
+        with pytest.raises(
+            ValidationError,
+            match="1 validation error for SlurmJobParams\nname\n  field required.*",
+        ):
+            get_job_parameters(slurm_parameters={})
+
+    def test_base_case__success(self):
+        """
+        Base case should succeed with a valid name.
+        """
+        desired_value = SlurmJobParams(name="test-test")
+
+        actual_value = get_job_parameters(slurm_parameters={}, name="test-test")
+
+        assert actual_value == desired_value
+
+    def test_priority(self):
+        """
+        Test that slurm parameters have priority over extra keyword arguments.
+        """
+        desired_value = SlurmJobParams(name="high-priority")
+
+        actual_value = get_job_parameters(
+            slurm_parameters=dict(name="high-priority"), name="test-test"
+        )
+
+        assert actual_value == desired_value
+
+    def test_extra_arguments(self):
+        """
+        Test that SlurmJobParams can be constructed with extra keyword arguments.
+        """
+        desired_value = SlurmJobParams(foo="bar", name="test-test")
+
+        actual_value = get_job_parameters(
+            slurm_parameters=dict(foo="bar"), name="test-test"
```

Comment:
I would like to see a test case that covers collision of slurm_parameters and kwargs.

Reply:
Do you mean like in the test above?

https://github.com/omnivector-solutions/cluster-agent/blob/8fa4e0580f4941a7c65cd33bf59fe91333853929/tests/jobbergate/test_submit.py#L445-L455

Reply:
Oh, jeez. Sorry I didn't read this thoroughly. 🤦